spec/models/infra_conversion_job_spec.rb

```diff
@@ -70,18 +343,88 @@
       it_behaves_like 'allows error signal'
 
       it_behaves_like 'doesn\'t allow start signal'
+      it_behaves_like 'doesn\'t allow collapse_snapshots signal'
     end
   end
 
-  context 'operations' do
-    let(:poll_interval) { Settings.transformation.job.retry_interval }
-
+  context 'transition methods' do
     context '#start' do
       it 'to poll_automate_state_machine when preflight_check passes' do
         expect(job).to receive(:queue_signal).with(:poll_automate_state_machine)
         job.signal(:start)
         expect(task.reload.state).to eq('migrate')
-        expect(task.options[:workflow_runner]).to eq('automate')
+      end
+    end
+
+    context '#collapse_snapshots' do
+      let(:async_task) { FactoryBot.create(:miq_task, :userid => user.id) }
+      let(:snapshot) { FactoryBot.create(:snapshot, :vm_or_template => vm_vmware) }
+
+      before do
+        job.state = 'collapsing_snapshots'
+        allow(MiqTask).to receive(:find).and_return(async_task)
+        # allow(MiqTask).to receive(:find).with(async_task.id).and_return(async_task)
+        allow(vm_vmware).to receive(:remove_all_snapshots_queue).with(user.id).and_return(async_task.id)
+      end
+
+      it 'abort_conversion when collapse_snapshots times out' do
+        job.context[:retries_collapsing_snapshots] = 960
+        expect(job).to receive(:update_migration_task_progress).once.ordered.with(:on_entry)
+        expect(job).to receive(:update_migration_task_progress).once.ordered.with(:on_error)
+        expect(job).to receive(:abort_conversion).with('Collapsing snapshots timed out', 'error')
+        job.signal(:collapse_snapshots)
+      end
+
+      it 'exits if async task does not exist and vm does not support remove_all_snapshots' do
+        expect(job).to receive(:update_migration_task_progress).once.ordered.with(:on_entry)
+        expect(job).to receive(:update_migration_task_progress).once.ordered.with(:on_exit)
+        job.signal(:collapse_snapshots)
+      end
+
+      it 'queues an async task and retries if async task does not exist and vm supports remove_all_snapshots' do
+        allow(vm_vmware).to receive(:snapshots).and_return([snapshot])
+        async_task.update!(:state => 'queued')
```

Comment:
I think you want `Queued` (capital Q). I changed it to `MiqTask::STATE_QUEUED`, but alas, that did not solve it.